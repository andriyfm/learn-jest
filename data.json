[
  {
    "id": 1,
    "title": "Sunt tempor pariatur labore elit occaecat tempor.",
    "description": "Adipisicing aute dolore id sint consectetur laborum in proident officia aliqua non non esse culpa. Enim duis ad consequat ipsum amet enim quis tempor esse Lorem elit sint excepteur. Proident nostrud enim aliqua sunt id laboris tempor qui enim sunt nisi labore officia."
  },
  {
    "id": 2,
    "title": "Nostrud consequat ad nulla tempor sit culpa proident.",
    "description": "Aliqua laborum in exercitation eiusmod non voluptate minim pariatur. Sit amet eu sint ad velit labore aliquip qui ea laboris eiusmod dolor. Duis in adipisicing fugiat cupidatat consectetur laboris tempor adipisicing nisi sit. Do enim nulla laborum proident enim cillum voluptate magna excepteur labore. Pariatur dolore reprehenderit voluptate nostrud proident ut cupidatat elit ullamco ut tempor in irure ullamco. Pariatur mollit deserunt id nulla occaecat excepteur aute occaecat ut."
  }
]
