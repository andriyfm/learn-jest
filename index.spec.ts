import data from './data.json'

test('the data is correct', () => {
    expect(data).toMatchSnapshot()
    expect(data).toHaveLength(2)
    expect(data.map(item => item.id)).toEqual([1,2])
})

data.forEach((el, i) => {
    test(`data ${i}: harus mempunyai property id, title & description`, () => {
        expect(el).toHaveProperty('id')
        expect(el).toHaveProperty('title')
        expect(el).toHaveProperty('description')
    })
});