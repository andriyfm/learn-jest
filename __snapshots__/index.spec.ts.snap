// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the data is correct 1`] = `
Array [
  Object {
    "description": "Adipisicing aute dolore id sint consectetur laborum in proident officia aliqua non non esse culpa. Enim duis ad consequat ipsum amet enim quis tempor esse Lorem elit sint excepteur. Proident nostrud enim aliqua sunt id laboris tempor qui enim sunt nisi labore officia.",
    "id": 1,
    "title": "Sunt tempor pariatur labore elit occaecat tempor.",
  },
  Object {
    "description": "Aliqua laborum in exercitation eiusmod non voluptate minim pariatur. Sit amet eu sint ad velit labore aliquip qui ea laboris eiusmod dolor. Duis in adipisicing fugiat cupidatat consectetur laboris tempor adipisicing nisi sit. Do enim nulla laborum proident enim cillum voluptate magna excepteur labore. Pariatur dolore reprehenderit voluptate nostrud proident ut cupidatat elit ullamco ut tempor in irure ullamco. Pariatur mollit deserunt id nulla occaecat excepteur aute occaecat ut.",
    "id": 2,
    "title": "Nostrud consequat ad nulla tempor sit culpa proident.",
  },
]
`;
